skbio/sequence/_sequence.py

```diff
@@ -187,30 +351,107 @@ def metadata(self):
     def metadata(self, metadata):
         if not isinstance(metadata, dict):
             raise TypeError("metadata must be a dict")
-        self._metadata = metadata
+        # shallow copy
+        self._metadata = metadata.copy()
+
+    @metadata.deleter
+    def metadata(self):
+        if hasattr(self, '_metadata'):
```

Comment:
`del` gets called from the constructor where `self._metadata` hasn't been assigned yet. I did it this way to have only a single representation of "missing" metadata (defined by the deleter). I have a comment about this in the constructor but I'll also add one here for clarity.